src/AxelarProxy.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import { AxelarExecutable } from "lib/axelar-gmp-sdk-solidity/contracts/executable/AxelarExecutable.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+
+contract AxelarProxy is AxelarExecutable {
+    using Address for address;
+
+    event LogicCallEvent(address target, bytes callData);
+
+    error AxelarProxy__WrongSource();
+    error AxelarProxy__NoTokens();
+
+    bytes32 public constant SOMMELIER_CHAIN_HASH = keccak256(bytes("sommelier"));
+
+    constructor(address gateway_) AxelarExecutable(gateway_) {}
+
+    /**
+     * @notice Execution logic.
+     * @dev Verifies message is from Sommelier, otherwise reverts.
```

Comment:
Please add note about how the `_execute()` function checks with the `Gateway` as well from the `Source` chain(?). Please confirm if that is how it works too, this was just from separate discussions with the protocol team, although I may have interpreted it incorrectly.

Reply:
We discussed this, and the `AxelarProxy.sol` `execute()` function has logic that lays this out.